news-catalyst/google-app-scripts
# Handle preview

Closes #200 for preview.

This was a little complicated and while I've tested it, I won't exactly be surprised if issues come up. That said, I did test it numerous times on a new document and then on two different documents repeatedly 🤞 

This PR (latest code in script editor) should:

* create a new article, article translation, article authors (rel), article tags (rel)
* or update an existing article, creating a new article translation, updating any author or tag relations

To test, 

* either open a document that I was using already (both are in the "test as add-on" list)
* or add a new test case using a document in the Tiny News Collective (shared google drive) > Testing > oaklyn > articles folder (there are 2 docs there that I haven't tried: kensington's last-minute gift guide and while harm reduction...)

You should see either "Retrieved article with ID x" or "article not found" messaging upon opening the sidebar.

Filling in the form and clicking preview should result in a preview link. 

I'll work on ensuring the preview link actually works tomorrow.
